DSCResources/MSFT_xIisFeatureDelegation/MSFT_xIisFeatureDelegation.psm1

```diff
@@ -0,0 +1,160 @@
+######################################################################################
+function Set-TargetResource
+{
+	param
+	(	
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$SectionName,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+    )
+
+    CheckIISPoshModule
+    [string]$oMode = GetOverrideMode $SectionName
+
+
+    if ($oMode -eq "Allow" -and $OverrideMode -eq "Deny")
+    {
+         Set-webconfiguration -Location "" -Filter "/system.webServer/$SectionName" -PSPath "machine/webroot/apphost" -metadata overrideMode -value Deny
+         Write-Verbose($($LocalizedData.ChangedMessage) -f $SectionName,"Deny")
+    }
+    elseif ($oMode -eq "Deny" -and $OverrideMode -eq "Allow")
+    {
+         Set-webconfiguration -Location "" -Filter "/system.webServer/$SectionName" -PSPath "machine/webroot/apphost" -metadata overrideMode -value Allow
+         Write-Verbose($($LocalizedData.ChangedMessage) -f $SectionName,"Allow")
+    }
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test if the given section has the required OverrideMode
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+	param
+	(	
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$SectionName,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+	)
+
+    CheckIISPoshModule
+
+    [bool]$DesiredConfigurationMatch = $true;
+    [string]$oMode = GetOverrideMode $SectionName
+
+    if ($oMode -eq $OverrideMode)
+    {
+        $DesiredConfigurationMatch = $true;
+    }
+    elseif ($oMode -ne $null -and $OverrideMode -ne $oMode)
+    {
+        $DesiredConfigurationMatch = $false;
+    }
+    else
+    {
+        Write-Verbose($($LocalizedData.ConfigNotFound) -f $SectionName)
+        $DesiredConfigurationMatch = $false;
+    }
+        
+	return $DesiredConfigurationMatch
+}
+
+Function CheckIISPoshModule
+{
+    # Check if WebAdministration module is present for IIS cmdlets
+    if(!(Get-Module -ListAvailable -Name WebAdministration))
+    {
+        Throw $LocalizedData.NoWebAdministrationModule
+    }
+}
+
+Function GetOverrideMode([string]$section)
+{
+    $errorMessage = $($LocalizedData.UnableToGetConfig) -f $section
+    if ((get-webconfiguration -Location "" -Filter /system.webServer/$section).count -eq 0)
+    {        
+        Throw $errorMessage;
+    }
+
+    [string]$oMode = ((get-webconfiguration -Location "" -Filter /system.webServer/$section -metadata).Metadata).effectiveOverrideMode
+
+    # check for a single value.
+    if ($oMode -notmatch "^(Allow|Deny)$")
+    {
+        Throw $errorMessage
```

Comment:
I added the check because at some point `$oMode` had a value of `Allow Allow` because I got my XPath wrong and returned multiple nodes. It shouldn't happen anymore, but I thought I can't hurt and kept it in there.